pkg/scheduler/framework/cycle_state_test.go

```diff
@@ -31,44 +32,120 @@ func (f *fakeData) Clone() StateData {
-		t.Errorf("clone failed, got %q, expected %q", v.data, data1)
+	if a == nil || b == nil {
+		return false, fmt.Sprintf("one CycleState is nil, but another one is not nil. A: %v, B: %v", a, b)
 	}
 
-	originalValue.data = data2
-	original, err := state.Read(key)
-	if err != nil {
-		t.Errorf("failed to read original value: %v", err)
+	if a.recordPluginMetrics != b.recordPluginMetrics {
+		return false, fmt.Sprintf("CycleState A and B have a different recordPluginMetrics. A: %v, B: %v", a.recordPluginMetrics, b.recordPluginMetrics)
 	}
-	if v, ok := original.(*fakeData); ok && v.data != data2 {
-		t.Errorf("original value should change, got %q, expected %q", v.data, data2)
+
+	var msg string
+	isEqual := true
+	countA := 0
+	a.storage.Range(func(k, v1 interface{}) bool {
+		countA++
+		v2, ok := b.storage.Load(k)
+		if !ok {
+			isEqual = false
+			msg = fmt.Sprintf("CycleState B doesn't have the data which CycleState A has. key: %v, data: %v", k, v1)
+			return false
+		}
+
+		typed1, ok1 := v1.(*fakeData)
+		typed2, ok2 := v2.(*fakeData)
+		if !ok1 || !ok2 {
+			isEqual = false
+			msg = fmt.Sprintf("CycleState has the data which is not type *fakeData.")
+			return false
+		}
+
+		if typed1.data != typed2.data {
+			isEqual = false
+			msg = fmt.Sprintf("CycleState B has a different data on key %v. A: %v, B: %v", k, typed1.data, typed2.data)
+			return false
+		}
+
+		return true
+	})
+
+	if !isEqual {
+		return false, msg
 	}
 
-	if v, ok := valueCopy.(*fakeData); ok && v.data != data1 {
-		t.Errorf("cloned copy should not change, got %q, expected %q", v.data, data1)
+	countB := 0
+	b.storage.Range(func(k, _ interface{}) bool {
+		countB++
+		return true
+	})
+
+	if countA != countB {
+		return false, fmt.Sprintf("two Cyclestates have different numbers of data. A: %v, B: %v", countA, countB)
 	}
+
+	return true, ""
 }
 
-func TestCycleStateCloneNil(t *testing.T) {
-	var state *CycleState
-	stateCopy := state.Clone()
-	if stateCopy != nil {
-		t.Errorf("clone expected to be nil")
+func TestCycleStateClone(t *testing.T) {
+	tests := []struct {
+		name            string
+		state           *CycleState
+		wantClonedState *CycleState
+	}{
+		{
+			name:            "clone with recordPluginMetrics true",
+			state:           createCycleStateWithFakeData("data", true),
+			wantClonedState: createCycleStateWithFakeData("data", true),
+		},
+		{
+			name:            "clone with recordPluginMetrics false",
+			state:           createCycleStateWithFakeData("data", false),
+			wantClonedState: createCycleStateWithFakeData("data", false),
+		},
+		{
+			name:            "clone with nil CycleState",
+			state:           nil,
+			wantClonedState: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			state := tt.state
+			stateCopy := state.Clone()
+
+			if isEqual, msg := isCycleStateEqual(stateCopy, tt.wantClonedState); !isEqual {
+				t.Errorf("unexpected cloned state: %v\n", msg)
```

Comment:
no need to append `\n` (similar as L147)

Reply:
Fixed.